scipy/integrate/_quadrature.py

```diff
@@ -662,6 +664,254 @@ def simpson(y, x=None, dx=1.0, axis=-1, even='avg'):
+            # sub_integrals_h1[i] = 0 or sub_integrals_h2[i] = 0)
+            (sub_integrals_h1[1:] + sub_integrals_h2[:-1]),
+            # Integral over last subinterval can only be calculated from formula for h2
+            sub_integrals_h2[-1:],
+        ],
+        axis=0,
+    )
+    res = np.cumsum(sub_integrals, axis=0)
+    return res
+
+
+def _cumulative_simpson_equal_intervals(
+    y: np.ndarray, dx: np.ndarray, axis: int
+) -> np.ndarray:
+
+    y = y.swapaxes(0, axis)
+    dx = dx.swapaxes(0, axis)
+
+    # Consider a quadratic interpolation over each set of 3 adjacent points,
+    # a, a+h1, a+h1+h2. The subinterval widths are h1 and h2
+    # (in this case, h1 = h2 = dx)
+
+    # Calculate integral over the h1 subintervals.
+    # Calculate for all but last subinterval of y.
+    sub_integrals_h1 = dx[:-1] / 3.0 * \
+        (5.0 * y[:-2] / 4.0 + 2.0 * y[1:-1] - y[2:] / 4.0)
+
+    # Calculate integral over the h2 subintervals.
+    # Calculate for all but first subinterval of y.
+    sub_integrals_h2 = dx[:-1] / 3.0 * \
+        (-y[:-2] / 4.0 + 2.0 * y[1:-1] + 5.0 * y[2:] / 4.0)
+
+    # Addition of above formulae gives Simpson's 1/3 rule, see
+    # https://en.wikipedia.org/wiki/Simpson%27s_rule#Composite_Simpson's_1/3_rule
+
+    res = _cumulatively_sum_simpson_subintegrals(
+        sub_integrals_h1,
+        sub_integrals_h2,
+    )
+    return res.swapaxes(0, axis)
+
+
+def _cumulative_simpson_unequal_intervals(
+    y: np.ndarray, x: np.ndarray, axis: int
+) -> np.ndarray:
+
+    y = y.swapaxes(0, axis)
+    x = x.swapaxes(0, axis)
+
+    d = np.diff(x, axis=0)
+    if np.any(d <= 0):
+        raise ValueError("Input x must be monotonically increasing.")
+    h1 = d[:-1]
+    h2 = d[1:]
+    y1 = y[:-2]
+    y2 = y[1:-1]
+    y3 = y[2:]
+
+    # Consider a quadratic interpolation over each set of 3 adjacent points,
+    # a, a+h1, a+h1+h2. The subinterval widths are h1 and h2
+
+    # Calculate integral over the h1 subintervals.
+    # Calculate for all but last subinterval of y.
+    sub_integrals_h1 = (
+        y1 * (2.0 * h1 + 3.0 * h2) / (h1 + h2) * h1 / 6.0
+        + y2 * (h1 + 3.0 * h2) * h1 / (6.0 * h2)
+        - y3 * (h1**3 / (6.0 * h2)) / (h1 + h2)
+    )
+
+    # Calculate integral over the h2 subintervals.
+    # Calculate for all but first subinterval of y.
+    sub_integrals_h2 = (
+        -y1 * (h2**3 / (6.0 * h1)) / (h1 + h2)
+        + y2 * (3.0 * h1 + h2) * h2 / (6.0 * h1)
+        + y3 * (3.0 * h1 + 2.0 * h2) / (h1 + h2) * h2 / 6.0
+    )
+
+    # Addition of above formulae gives Simpson's 1/3 rule for unequal intervals, see
+    # https://en.wikipedia.org/wiki/Simpson%27s_rule#Composite_Simpson's_rule_for_irregularly_spaced_data
+
+    res = _cumulatively_sum_simpson_subintegrals(
+        sub_integrals_h1,
+        sub_integrals_h2,
+    )
+    return res.swapaxes(0, axis)
+
+
+def cumulative_simpson(y, x=None, dx=1.0, axis=-1, initial=None):
```

Comment:
Use some keyword only to simplify future maintenance.
```suggestion
def cumulative_simpson(y, *, x=None, dx=1.0, axis=-1, initial=None):
```

Reply:
Incorporated in [be6e1d7](https://github.com/scipy/scipy/pull/18151/commits/be6e1d75e7863d46c297ad24a9929392873dcf3a)